src/main/java/httpServer/ListenerAndResponder.java

```diff
@@ -40,46 +40,102 @@ public Socket keepListening() throws IOException {
     }
 
     public void readClientInput(Socket clientSocket) throws IOException {
-
+        System.err.println("Starting to read");
         ClientReadable bufferedReader = clientReaderFactory.makeReader(clientSocket);
-
-        String httpRequest;
-        httpRequest = readAllLines(bufferedReader);
+        HTTPRequest httpRequest;
+        httpRequest = readHeaderAndBodyAndReturnHttpRequest(bufferedReader);
```

Comment:
I think I pointed out in my last review that using `and` in your method or class names is probably a violation of single responsibility principle. In this case, it doesn't necessarily seem like a violation but I would recommend renaming this method anyway. What is its responsibility?

Reply:
As an additional note, you can just combine lines 45-46 in one line:
```java
HTTPRequest httpRequest = readHeaderAndBodyAndReturnHttpRequest(bufferedReader);
```

Reply:
It turns the String that get sent into a HTTPRequest object, so I renamed accordingly